drivers/nvdimm/pmem.c

```diff
@@ -45,11 +65,352 @@ static struct nd_region *to_region(struct pmem_device *pmem)
 	return to_nd_region(to_dev(pmem)->parent);
 }
 
+static bool pmem_dma_feq(struct dma_chan *chan, void *data)
+{
+	return dev_to_node(chan->device->dev) == (long)data;
+}
+
+static bool pmem_dma_fne(struct dma_chan *chan, void *data)
+{
+	return dev_to_node(chan->device->dev) != (long)data;
+}
+
+static void pmem_dma_init(struct pmem_device *pmem)
+{
+	struct device *dev = to_dev(pmem);
+	dma_cap_mask_t dma_mask;
+
+	dma_cap_zero(dma_mask);
+	dma_cap_set(DMA_MEMCPY, dma_mask);
+
+	/*
+	 * Prefer to allocate NUMA-local DMA channel for remote PMEM writes.
+	 * Remote writes are slower than local and need all boost we can give.
+	 */
+	pmem->rdma_chan = dma_request_channel(dma_mask, pmem_dma_feq,
+	    (void *)(long)pmem->rnode);
```

Comment:
Couldn't line 93 also be included in the conditional at line 94, and get rid of line 92 ? Those conditionals look identical to me 🤔

Reply:
The second channel request may fail as well.

Reply:
👍🏼